pgjdbc/src/main/java/org/postgresql/ds/common/BaseDataSource.java

```diff
@@ -124,22 +124,49 @@ public void setLogWriter(PrintWriter printWriter) {
    * Gets the name of the host the PostgreSQL database is running on.
    *
    * @return name of the host the PostgreSQL database is running on
+   * @deprecated use {@link #getServerNames()}
    */
+  @Deprecated
   public String getServerName() {
-    return serverName;
+    return serverNames[0];
+  }
+
+  /**
+   * Gets the name of the host(s) the PostgreSQL database is running on.
+   *
+   * @return name of the host(s) the PostgreSQL database is running on
+   */
+  public String[] getServerNames() {
+    return serverNames;
   }
 
   /**
    * Sets the name of the host the PostgreSQL database is running on. If this is changed, it will
    * only affect future calls to getConnection. The default value is <tt>localhost</tt>.
    *
    * @param serverName name of the host the PostgreSQL database is running on
+   * @deprecated use {@link #setServerNames(String[])}
    */
+  @Deprecated
   public void setServerName(String serverName) {
```

Comment:
The body of this function should invoke the array version so that the logic is in one place and any future changes happen once:

```java
public void setServerName(String serverName) {
    this.setServerName(new String[] { serverName });
}
```

Reply:
Typo on my part, should be `setServerNames(...)` in the code block.

Reply:
@sehrope are you happy with this? Can you resolve the conversation if so /